dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/validation/OASValidator.java

```diff
@@ -80,6 +89,21 @@ public boolean addOperationId(String operationId) {
         return !operationIds.add(operationId);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public void addLinkOperationId(String operationId, String location) {
+        linkOperationIds.put(operationId, location);
```

Comment:
multiple Link objects could refer to the same operationId, in which case only one validation event will be produced when the operationId is not valid (for the last link location that was added). Instead of `Map<String, String>`, use `Map<String, Set<String>>` for linkOperationIds. When trying to adding an item to the Map, see if the key already exists, if so then get the Set and add the location to it. Otherwise, create a Set, add location to it and add the Set as a value. That way we can gather all Link locations that referred an invalid operationId.